src/ocnews_api.cpp

```diff
@@ -18,12 +21,45 @@ typedef std::unique_ptr<json_object, decltype(*json_object_put)> json_uptr;
 typedef std::unique_ptr<CURL, decltype(*curl_easy_cleanup)> curl_uptr;
 
 ocnews_api::ocnews_api(configcontainer* c) : remote_api(c) {
-	auth = cfg->get_configvalue("ocnews-login")+":"+cfg->get_configvalue("ocnews-password");
+	std::string user = cfg->get_configvalue("ocnews-login");
+	bool flushed = false;
+
+	if (user == "") {
+		std::cout << std::endl;
+		std::cout.flush();
+		flushed = true;
+		std::cout << "Username for ocNews: ";
+		std::cin >> user;
+
+		if (user.empty())
+			LOG(level::CRITICAL, "ocnews_api::ocnews_api: No user set");
+	}
+
+	std::string pass = cfg->get_configvalue("ocnews-password");
+	std::string pass_file = cfg->get_configvalue("ocnews-passwordfile");
+	std::string pass_eval = cfg->get_configvalue("ocnews-passwordeval");
+	if ( pass == "" ) {
+		if ( pass_file != "" ) {
+			pass = read_password(pass_file);
+		} else if ( pass_eval != "") {
+			pass = eval_password(pass_eval);
+		} else {
+			if (!flushed) {
+				std::cout << std::endl;
+				std::cout.flush();
+			}
+			// Find a way to do this in C++ by removing cin echoing.
+			pass = std::string( getpass("Password for ocNews: ") );
+		}
+
+		if (pass.empty())
+			LOG(level::CRITICAL, "ocnews_api::ocnews_api: No password set");
```

Comment:
I don't know why the author of ocNews class chose to log an error and proceed instead of failing the auth; let's not do that. Can you just take the code you already have in, say, `ttrss_api.cpp`, stick it into a method in `remote_api.cpp`, and re-use it in all remote APIs?